router/api.go

```diff
@@ -187,7 +196,29 @@ func (api *AWSServerlessApi) Swagger() ([]byte, error) {
 	}
 
 	return nil, fmt.Errorf("no swagger definition found")
+}
 
+func (api *AWSServerlessApi) ensureJSON(data []byte) ([]byte, error) {
+	var tmpDefinition interface{}
+	err := json.Unmarshal(data, &tmpDefinition)
+
+	if err != nil { // may be yaml
+		err = yaml.Unmarshal(data, &tmpDefinition)
+
+		if err != nil {
+			// we can't make it work either as json or yaml. fail :(
```

Comment:
nit, but at this point you've only tested it's not yaml right?

Reply:
I've first tested that it' not JSON at line 203. if I also fail at 206 then I cannot unmarhsal it as either.

Reply:
Gotcha.